pandas/_libs/testing.pyx

```diff
@@ -143,11 +143,9 @@ cpdef assert_almost_equal(a, b,
                 from pandas.util.testing import assert_attr_equal
                 assert_attr_equal('dtype', a, b, obj=obj)
 
-            try:
-                if array_equivalent(a, b, strict_nan=True):
-                    return True
-            except:
-                pass
+            if array_equivalent(a, b, strict_nan=True):
+                return True
+
```

Comment:
need to look if we should return False on 148

Reply:
Looking at this more closely, I've decided it should be handled separately.  (also there are some optimizations to be made in that separate pass).  So this should be good to go.